compiler-core/src/language_server/engine.rs

```diff
@@ -360,7 +359,12 @@ where
                 if !type_.public {
                     continue;
                 }
-                completions.push(type_completion(Some(&alias), name, type_));
+                let module = match import.clone().as_name {
+                    Some((AssignName::Variable(name), _)) => Some(name),
+                    Some((AssignName::Discard(_), _)) => None,
+                    None => Some(module.name.clone()),
```

Comment:
This is registering the name of the module in the scope, but it should instead be the used name.

```rust
let Some(alias) = import.used_name() else {
    continue;
}

for (name, type_) in &module.types {
    if !type_.public {
        continue;
    }
    completions.push(type_completion(Some(&alias), name, type_));
}
```